filebeat/input/logv2/input.go

```diff
@@ -0,0 +1,182 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logv2
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/v7/filebeat/channel"
+	v1 "github.com/elastic/beats/v7/filebeat/input"
+	"github.com/elastic/beats/v7/filebeat/input/filestream"
+	loginput "github.com/elastic/beats/v7/filebeat/input/log"
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/beats/v7/libbeat/features"
+	"github.com/elastic/beats/v7/libbeat/management"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/go-concert/unison"
+)
+
+const pluginName = "log"
+
+func init() {
+	// Register an input V1, to replace the Log input one.
+	if err := v1.Register(pluginName, newV1Input); err != nil {
+		panic(err)
+	}
+}
+
+// runAsFilestream validates cfg as a Log input configuration, if it is
+// valid, then checks whether the configuration should be run as
+// Filestream input. On any error the boolean value must be ignore and
+// no input started. runAsFilestream also sets the input type accordingly.
+func runAsFilestream(cfg *config.C) (bool, error) {
+	// First of all, ensure the Log input configuration is valid.
+	// This ensures we return configuration errors compatible
+	// with the log input.
+	if err := loginput.IsConfigValid(cfg); err != nil {
+		return false, err
+	}
+
+	// Global feature flag that forces all Log input instances
+	// to run as Filestream.
+	if features.LogInputRunFilestream() {
+		return true, nil
+	}
+
+	// Only allow to run the Log input as Filestream if Filebeat
+	// is running under Elastic Agent.
+	if !management.UnderAgent() {
+		return false, nil
+	}
```

Comment:
Shouldn't it be before the `features.LogInputRunFilestream()` or together, `features.LogInputRunFilestream() && management.UnderAgent()`?

Reply:
Isn't the previous comment:
> 	// Global feature flag that forces all Log input instances
	// to run as Filestream.

enough to explain that the feature flag takes precedence over everything?


Does this version sound better to you:
> 	// Global feature flag that forces all Log input instances
	// to run as Filestream, even when not running under
	// Elastic Agent

?